filter/specItemFilter.go

```diff
@@ -233,11 +233,44 @@ func validateTagExpression(tagExpression string) {
 
 func filterSpecsByScenarioName(specs []*gauge.Specification, scenariosName []string) []*gauge.Specification {
 	filteredSpecs := make([]*gauge.Specification, 0)
+	scenarios := filterValidScenarios(specs, scenariosName)
 	for _, spec := range specs {
-		spec.Filter(newScenarioFilterBasedOnName(scenariosName))
+		spec.Filter(newScenarioFilterBasedOnName(scenarios))
 		if len(spec.Scenarios) != 0 {
 			filteredSpecs = append(filteredSpecs, spec)
 		}
 	}
 	return filteredSpecs
 }
+
+func filterValidScenarios(specs []*gauge.Specification, headings []string) []string {
+	filteredScenarios := make([]string, 0)
+	allScenarios := GetAllScenarios(specs)
+	for _, heading := range headings {
+		if isContains(allScenarios,heading) {
+			filteredScenarios = append(filteredScenarios,heading)
+		} else {
+			logger.Warningf(true,"Warning: scenario name - \"%s\" not found",heading)
+		}
+	}
+	return filteredScenarios;
+}
+
+func isContains(scenarios []string, heading string) bool {
```

Comment:
`isContains` => `exists`

this can be a local function if it isnt used elsewhere

Reply:
it's a local function, start with the small letter. Do you want to rename this?

Reply:
I meant something like 

```
var exists = func(headings []string, heading string) bool {
    for _, h := range headings {
        if h == heading {
            return true
        }
    }
    return false
}
```

inside the function `filterValidScenarios`

Reply:
it's cleaner than do everything in a single function. I think it's a good way of doing that. Does something wrong with this?